pages/design/rule-design.md

```diff
@@ -2,29 +2,73 @@
 
-6. **[Expectations](#expectations)**: Assert what must be true about the target elements, in order for them to pass the rule.
+6. **[Background](#background)**: Provide information on relevant resources referenced when developing the rule.
+
+7. **[Test cases](#test-cases)**: Define a range of code examples that demonstrate pass, fail and inapplicable outcomes for readers and for validating implementations.
 
 ## Rule Properties
 
-### Rule name and identifier
+A rule file should have a unique hyphenated filename, created from condensing the name and the unique identifier of the rule, _e.g. "aria-attr-defined-5f99a7"_. A rule file should also have a number of metadata properties at the top of the file.
+
+For more details, see [ACT Rules Format: Structure](https://www.w3.org/TR/act-rules-format/#act-rule-structure).
+
+### Rule identifier
+
+The 6-digit identifier is created using a random hex generator, such as https://www.browserling.com/tools/random-hex. The identifier remains the same if the rule name changes.
+
+### Rule name
+
+The agreed conventions for naming of rules are as follows:
+
+- Be succinct, direct and declarative, and avoid unnecessary words, _e.g. an, the_.
+- Use declarative verbs, _e.g. has, have, is_.
+- Use back ticks to indicate words from code, _e.g. `label` v. Label_.
+- Identify code as an element or attribute as appropriate, _e.g. `title` element, `title` attribute_.
+- Use sentence case, _e.g. "Heading is descriptive"_, unless using code, _e.g. "`id` attribute value is unique"_, or initials, _e.g. "HTML page has title"_. Note: names starting with a backtick also need to be wrapped with quotes for the parser.
+- Use singular tense so that necessary plurals stand out, _e.g. "`video` element has captions"_.
+- Front load the applicable thing, _e.g. "Button has accessible name"_.
+- Refer to each applicable thing in a consistent way across rules, especially when related or checking the same applicable thing.
+- Describe the passing condition, _e.g. "`id` attribute value is unique", rather than "page has no duplicate `id`s"_.
+- Do not use hyphens or dashes unless correct for code, _e.g. `aria-*`_.
+- Do not use camelCase unless correct for code, _e.g. `iframe` not iFrame, `feImage` in SVG_.
+- If unsure, refer to WCAG language used.
+- Aim to be unique, which should happen if title includes the applicable thing and passing condition.
+
+### Rule type
+
+The rule type is either atomic or composite.
+
+For more details, see [ACT Rules Format: Rule Types](https://www.w3.org/TR/act-rules-format/#rule-types).
 
-The rule must have a unique name.
+### Description
 
-### Test aspects
+A brief description of what the rule does. This should be declarative rather than negative. It should begin with "This rule checks that..." or similar. It should use plain language, and be a well-formed grammatically correct sentence that finishes with a full stop/period.
```

Comment:
```suggestion
The description is a brief synopsis of what the rule does. This should be declarative rather than negative. It should begin with "This rule checks that..." or similar. It should use plain language, and be a well-formed grammatically correct sentence that finishes with a full stop/period.
```